### DISCUSSION TITLE
Fatal error in CalculateSelectViewValue.php

### DISCUSSION THREAD

**Discussion by DaveMaks** [totumonline/totum-ru-issues-and-discussions#323] [ОШБК!] [answered]
Доброе время суток, Алексей, 
Есть необходимость добавить проверку на is_array($rows), так как при случайном добавлении к полю типа _Селект_ в **Код селекта** функции =:selectList(...), приложение вываливается в ошибку 

> `Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, int given in /var/www/totum-mit/totum/common/calculates/CalculateSelectViewValue.php:59 Stack trace: #0 /var/www/totum-mit/totum/common/calculates/CalculateSelectViewValue.php(59): array_key_exists() #1 /var/www/totum-mit/totum/common/calculates/CalculateSelect.php(38): totum\common\calculates\CalculateSelectViewValue->getPreparedList() #2 /var/www/totum-mit/totum/fieldTypes/Select.php(453): totum\common\calculates\CalculateSelect->exec() #3 /var/www/totum-mit/totum/fieldTypes/Select.php(596): totum\fieldTypes\Select->calculateSelectViewList() #4 /var/www/totum-mit/totum/tableTypes/aTable.php(1510): totum\fieldTypes\Select->addViewValues() #5 /var/www/totum-mit/totum/moduls/Table/ReadTableActions.php(1291): totum\tableTypes\aTable->getValuesAndFormatsForClient() #6 /var/www/totum-mit/totum/moduls/Table/ReadTableActions.php(1337): totum\moduls\Table\ReadTableActions->addValuesAndFormatsOfParams() #7 /var/www/totum-mit/totum/moduls/Table/ReadTableActions.php(1200): totum\moduls\Table\ReadTableActions->getTableClientData() #8 /var/www/totum-mit/totum/moduls/Table/TableController.php(556): totum\moduls\Table\ReadTableActions->getFullTableData() #9 /var/www/totum-mit/totum/common/controllers/Controller.php(47): totum\moduls\Table\TableController->actionTable() #10 /var/www/totum-mit/totum/moduls/Table/TableController.php(534): totum\common\controllers\Controller->__actionRun() #11 /var/www/totum-mit/totum/common/controllers/WithAuthTrait.php(18): totum\moduls\Table\TableController->__actionRun() #12 /var/www/totum-mit/totum/moduls/Table/TableController.php(504): totum\moduls\Table\TableController->__run() #13 /var/www/totum-mit/http/index.php(45): totum\moduls\Table\TableController->doIt() #14 {main} thrown in /var/www/totum-mit/totum/common/calculates/CalculateSelectViewValue.php on line 59`

после чего нет возможности откатить изменения.

https://github.com/totumonline/totum-mit/blob/fdc20c02a1ef6f894f70b20cf18674aee2828753/totum/common/calculates/CalculateSelectViewValue.php#L59

**Comment by erpic-ru** [comment, marked as answer]
Зайдите через Системные таблицы — Основные — Таблицы и поля — Состав таблиц и поправьте нужный код селекта в нужном поле

https://ru.docs.totum.online/debugging

**Reply by DaveMaks** [reply]
> Зайдите через Системные таблицы — Основные — Таблицы и поля — Состав таблиц и поправьте нужный код селекта в нужном поле
> 
> https://ru.docs.totum.online/debugging

Это просто на будущее чтобы приложение в глобальный exception не вываливалось)

**Reply by ttm-alexeypolunin** [reply]
Спс, поправим

**Reply by ttm-alexeypolunin** [reply]
Посмотрели. В декабре фиксили это место — обновитесь.
